digdag-standards/src/main/java/io/digdag/standards/operator/PyOperatorFactory.java

```diff
@@ -101,59 +104,130 @@ public TaskResult runTask()
                 .build();
         }
 
-        private Config runCode(Config params)
+        private Config runCode()
+                throws IOException, InterruptedException
+        {
+            final Config params = request.getConfig()
+                    .mergeDefault(request.getConfig().getNestedOrGetEmpty("py"));
+            final Config stateParams = TaskState.of(request).params();
+            final Path projectPath = workspace.getProjectPath(); // absolute
+
+            final CommandStatus status;
+            if (!stateParams.has("commandStatus")) {
+                // Run the code since command state doesn't exist
+                status = runCode(params, projectPath);
+            }
+            else {
+                // Check the status of the code running
+                final ObjectNode previousStatusJson = stateParams.get("commandStatus", ObjectNode.class);
+                status = checkCodeState(params, projectPath, previousStatusJson);
+            }
+
+            if (status.isFinished()) {
+                final int statusCode = status.getStatusCode();
+                if (statusCode != 0) {
+                    // Remove the polling state after fetching the result so that the result fetch can be retried
+                    // without resubmitting the code.
+                    stateParams.remove("commandStatus");
+                    throw new RuntimeException("Python command failed with code " + statusCode);
+                }
+
+                final Path workspacePath = workspace.getPath();
+                final Path outputPath = workspacePath.resolve(status.getIoDirectory()).resolve("output.json");
+                try (final InputStream in = Files.newInputStream(outputPath)) {
+                    return mapper.readValue(in, Config.class);
+                }
+            }
+            else {
+                stateParams.set("commandStatus", status);
+                throw TaskExecutionException.ofNextPolling(scriptPollInterval, ConfigElement.copyOf(stateParams));
+            }
+        }
+
+        private CommandStatus runCode(final Config params, final Path projectPath)
                 throws IOException, InterruptedException
         {
-            String inFile = workspace.createTempFile("digdag-py-in-", ".tmp");
-            String outFile = workspace.createTempFile("digdag-py-out-", ".tmp");
+            final Path tempDir = workspace.createTempDir(String.format("digdag-py-%d-", request.getTaskId()));
+            final Path workingDirectory = workspace.getPath(); // absolute
+            final Path inputPath = tempDir.resolve("input.json"); // absolute
+            final Path outputPath = tempDir.resolve("output.json"); // absolute
```

Comment:
can we make this `"output.json"` literal at least to be a constant so that we can let compilers check typo?

Reply:
Thank you for suggesting. https://github.com/treasure-data/digdag/pull/835/commits/20a374ec23f193207f239c76a68745a62e088c13